pegen/pegen.c

```diff
@@ -381,26 +381,76 @@ number_token(Parser *p)
     return Constant(c, NULL, t->lineno, t->col_offset, t->end_lineno, t->end_col_offset, p->arena);
 }
 
+static int
+parsestr(Parser* p, const char* s, int *bytesmode, int *rawmode,
+         PyObject **result, const char **fstr, Py_ssize_t *fstrlen);
+
 expr_ty
 string_token(Parser *p)
 {
     Token *t = expect_token(p, STRING);
-    if (t == NULL)
-        return NULL;
-    char *s = NULL;
-    Py_ssize_t len = 0;
-    if (PyBytes_AsStringAndSize(t->bytes, &s, &len) < 0)
+    PyObject *u_kind = NULL;
```

Comment:
You can move this to line 439. (I like declaring variables near the first point they're needed.)